### PR TITLE
:sparkles: Adding aws-cli to docker image

### PR DESCRIPTION
**Pull request by jaswalkiranavtar**
## Summary
This image will be use as initContainer to copy the aws-cli to klusterlet-agent when initialized with awsirsa auth type.
This will allow us to keep registration-operator docker image clean.

## Related issue(s)

Fixes # https://github.com/open-cluster-management-io/ocm/issues/514

### CODE REVIEW

File: build/Dockerfile.aws-cli [hunk 1]
```diff
@@ -0,0 +1,25 @@
+FROM golang:1.22-bullseye AS builder
+ARG OS=linux
+ARG ARCH=amd64
+
+# Downloading aws-cli
+WORKDIR /tmp
+RUN apt-get update
+RUN apt-get install unzip
+RUN unzip -v
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
```
Comment (qiujian16) [inline build/Dockerfile.aws-cli:10]: can we make this as arch agnostic? we would also need to support other arch.

**Comment by qiujian16** [comment]
also would you help to remind me on why we do not use aws go sdk? https://pkg.go.dev/github.com/aws/aws-sdk-go-v2. I remember we discussed this before. It will be good to leave some comments so we do not miss the reason.

**Comment by jaswalkiranavtar** [comment]
> also would you help to remind me on why we do not use aws go sdk? https://pkg.go.dev/github.com/aws/aws-sdk-go-v2. I remember we discussed this before. It will be good to leave some comments so we do not miss the reason.

Yes, that is because the eks kubeconfig uses aws cli. We discussed about it and @deads2k suggested an option to copy aws cli from init container to klusterlet agent container. With this we do not have to maintain 2 copies of registration operator docker image, one for csr and other for aws auth.

https://github.com/open-cluster-management-io/enhancements/pull/125#discussion_r1678003208

With

File: build/Dockerfile.aws-cli [hunk 2]
```diff
@@ -0,0 +1,25 @@
+FROM golang:1.22-bullseye AS builder
+ARG OS=linux
+ARG ARCH=x86_64
```
Comment (jaswalkiranavtar) [inline build/Dockerfile.aws-cli:3]: # For ARM
ARG ARCH=aarch64

File: build/Dockerfile.aws-cli (hunk 1 above)
Comment (jaswalkiranavtar) [reply build/Dockerfile.aws-cli]: Pushed the change, please advise if it looks ok. Will fix the commit sign-off tomorrow moning.

Comment (qiujian16) [reply build/Dockerfile.aws-cli]: looks good

**Comment by qiujian16** [comment]
/approve

@mikeshng I think we also need to create a repo in quay for this image.

**Comment by mikeshng** [comment]
> /approve
> 
> @mikeshng I think we also need to create a repo in quay for this image.

Ok. I have submitted the repo creation request. Thank you all for your patience.

**Review by mikeshng — APPROVED** [review]
/lgtm

repo has been created `https://quay.io/repository/open-cluster-management/aws-cli`

**Pull request merged**